zcash-issue-dag.py

```diff
@@ -245,88 +267,168 @@ def chunks(lst, n):
 
     return ret
 
+def fetch_workspace_graph(op, workspace_id, repos, cursor):
+    dependencies = op.workspace(id=workspace_id).issue_dependencies(
+        # TODO: This causes a 500 Internal Server Error. We need the ZenHub repo IDs here,
+        # not the GitHub repo IDs (which the previous REST API used).
+        # repository_ids=repos,
+        first=100,
+        after=cursor,
+    )
+    dependencies.nodes.id()
+    dependencies.nodes.blocked_issue.number()
+    dependencies.nodes.blocked_issue.repository.gh_id()
+    dependencies.nodes.blocking_issue.number()
+    dependencies.nodes.blocking_issue.repository.gh_id()
+    dependencies.page_info.has_next_page()
+    dependencies.page_info.end_cursor()
+
+def get_dependency_graph(endpoint, workspace_id, repos):
+    edges = []
+    cursor = None
+
+    while True:
+        op = Operation(zenhub_schema.Query)
+        fetch_workspace_graph(op, workspace_id, repos, cursor)
 
-class ZHDepsResourceHandler(drest.resource.ResourceHandler):
-    def get(self, repo_id):
-        path = '/repositories/%d/dependencies' % repo_id
+        d = endpoint(op)
+        data = (op + d)
+
+        dependencies = data.workspace.issue_dependencies
+        edges += [
+            (
+                (node.blocking_issue.repository.gh_id, node.blocking_issue.number),
```

Comment:
`gh_id` is used as a method, not a property, above.

Reply:
Oh, that's exactly what it does: https://sgqlc.readthedocs.io/en/latest/sgqlc.operation.html#sgqlc.operation.Selector

Sigh. (They could have avoided needing to do the call.)